cfme/utils/template/README.md

```diff
@@ -0,0 +1,52 @@
+Available providers:
+
+    ec2
+    gce
+    openstack
+    scvmm
+    virtualcenter
+
+
+TESTED ON SUCH ENVIRONMENTS:
+
+    rhos11
+    rhos8
+
+
+Required arguments:
+
+    --provider PROVIDER_NAME
+        OR
+    --provider-type PROVIDER_TYPE
+        IGNORED WHEN --provider is used
+
+Optional arguments:
+
+    --stream CUSTOM_STREAM_NAME
+        REQUIRES --stream-url if stream name is not listed in cfme_data
+
+    --stream-url CUSTOM_URL_TO_STREAM_DIRECTORY
+        REQUIRES --stream
+
+    --image-url URL_PATH_TO_IMAGE_FILE
+
+    --template-name CUSTOM_TEMPLATE_NAME
+        if template name is used -> template will be formatted as {template}-{stream}
+    
+    --provider-data  # PATH TO LOCAL YAML FILE WITH LOCAL PROVIDER DATA
```

Comment:
I think we can greatly simplify a  bunch of the logic/methods throughout if we drop support for provider-data.

With support of `--provider`, its trivial to have your data defined in cfme_data.local.yaml or cfme_data.yaml.

Was there any driver that you found to keep support for `--provider-data`?

Reply:
No, there was no specific reason for keeping this support. So if we can just through it out, I'll do it.

Reply:
go for it